weatherman.py

```diff
@@ -0,0 +1,246 @@
+        print("Highest: " + str(temperature) + "C on " + str(month) + " " + str(day))
+
+    @staticmethod
+    def minimum_temperature(year):
+        temperature, day, month = Computations.minimum_temperature(year)
+        print("Lowest: " + str(temperature) + "C on " + str(month) + " " + str(day))
+
+    @staticmethod
+    def maximum_humidity(year):
+        humidity, month, day = Computations.maximum_humidity(year)
+        print("Humidity: " + str(humidity) + "% on " + str(month) + " " + str(day))
+
+    @staticmethod
+    def average_highest_temperature(year, month):
+        average = Computations.average_highest_temperature(year, month)
+        print("Highest Average: " + str(average) + "C")
+
+    @staticmethod
+    def average_lowest_temperature(year, month):
+        average = Computations.average_lowest_temperature(year, month)
+        print("Average Mean Humidity: " + str(average) + "%")
+
+    @staticmethod
+    def average_mean_humidity(year, month):
+        average = Computations.average_mean_humidity(year, month)
+        print("Lowest Average: " + str(average) + "C")
+
+    @staticmethod
+    def generate_barchart(temp, is_highest):
+        if is_highest:
+            return "\033[91m {}\033[00m".format("+" * temp)
+        return "\033[94m {}\033[00m".format("+" * temp)
+
+    @staticmethod
+    def maxmin_temperature_on_days(year, month):
+        for day in __weather__["Murree"][year][month]:
+            temp = Computations.maximum_temperature_on_day(year, month, day)
+            temp_ = Computations.minimum_temperature_on_day(year, month, day)
+            if temp:
+                print(str(day) + " " + Report.generate_barchart(temp, True) + " " + str(temp) + "C")
+            if temp_:
+                print(str(day) + " " + Report.generate_barchart(temp_, False) + " " + str(temp_) + "C")
+
+    @staticmethod
+    def maxmin_temperature_on_days_bonus(year, month):
+        for day in __weather__["Murree"][year][month]:
+            temp = Computations.maximum_temperature_on_day(year, month, day)
+            temp_ = Computations.minimum_temperature_on_day(year, month, day)
+            if temp and temp_:
+                print(str(day) + " " + Report.generate_barchart(temp_, False) + Report.generate_barchart(temp, True)
+                      + " " + str(temp_) + "C" + " " + str(temp) + "C")
+            elif temp:
+                print(str(day) + " " + Report.generate_barchart(temp, True) + " " + str(temp) + "C")
+            elif temp_:
+                print(str(day) + " " + Report.generate_barchart(temp_, False) + " " + str(temp_) + "C")
+
+
+if __name__ == '__main__':
+    # Command Line Arguments
+    # first: path of weather-files, if weather-files are in the same directory of python file just type None
+    # second: operation (-e, -a, -c)
+    # third: argument : year, month etc
+    if len(sys.argv) < 4:
+        print("Less no of Arguments")
```

Comment:
Does this approach seems good? We are having a validation on number instead of looking for what possible args we can have and what to do against them and anything other than that could be highlighted?
Rethink on this approach.